tsdb/engine/tsm1/encoding.go

```diff
@@ -580,22 +622,23 @@ func DecodeStringBlock(block []byte, tdec *TimeDecoder, vdec *StringDecoder, a *
 	return (*a)[:i], nil
 }
 
-func packBlockHeader(blockType byte) []byte {
-	return []byte{blockType}
-}
-
-func packBlock(ts []byte, values []byte) []byte {
+func packBlock(buf []byte, typ byte, ts []byte, values []byte) []byte {
 	// We encode the length of the timestamp block using a variable byte encoding.
 	// This allows small byte slices to take up 1 byte while larger ones use 2 or more.
-	b := make([]byte, 10)
-	i := binary.PutUvarint(b, uint64(len(ts)))
+	if len(buf) < 1+10+len(ts)+len(values) {
+		buf = make([]byte, 1+10+len(ts)+len(values))
```

Comment:
Would there a benefit in checking the capacity of `buf` instead, and unconditionally reslicing to `buf[:1+10+len(ts)+len(values)]` after potentially creating a new slice here?